Code/tests/test_filefinder.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+
+import plistlib
+import StringIO
+import unittest
+
+import mock
+from autopkglib import ProcessorError
+from autopkglib.FileFinder import FileFinder
+
+
+class TestFileFinder(unittest.TestCase):
+    """Test class for FileFinder Processor."""
+
+    def setUp(self):
+        self.good_env = {"find_method": "glob", "pattern": "test"}
+        self.bad_env = {"find_method": "fake"}
+        self.input_plist = StringIO.StringIO(plistlib.writePlistToString(self.good_env))
+        self.processor = FileFinder(infile=self.input_plist)
+
+    def tearDown(self):
+        self.input_plist.close()
+
+    def test_raise_if_not_glob(self):
+        """Raise an exception if glob is not passed to find_method."""
+        self.processor.env = self.bad_env
+        with self.assertRaises(ProcessorError):
+            self.processor.main()
+
+    @mock.patch("autopkglib.FileFinder.globfind")
+    def test_no_fail_if_good_env(self, mock_glob):
```

Comment:
I chose FileFinder to exemplify this because it's an easy example, but if there's a processor that actually makes changes to things on disk - such as something like PkgBuilder, FileMover, or Copier, etc. - all of those things have to be mocked so that the unit test itself doesn't actually do anything in production. 

So this is an easy example, but unfortunately not be a comprehensive enough example.